internal/v2/tls_config_store/tls_config_store.go

```diff
@@ -22,53 +25,217 @@ var (
+				MechanismOneof: &s2av2pb.AuthenticationMechanism_Token{"token"},
+			},
+		},
+		ReqOneof: &s2av2pb.SessionReq_GetTlsConfigurationReq {
+			&s2av2pb.GetTlsConfigurationReq {
+				ConnectionSide: commonpb.ConnectionSide_CONNECTION_SIDE_CLIENT,
+			},
+		},
+	}); err != nil {
+		return nil, err
+	}
+
+	// Get the response containing config from S2Av2.
+	resp, err := cstream.Recv()
 	if err != nil {
-		log.Fatalf("Failed to generate X509KeyPair: %v", err)
+		return nil, err
+	}
+
+	// TODO(rmehta19): Handle resp.GetStatus().
+
+	// Extract TLS configiguration from SessionResp.
+	tlsConfig := resp.GetGetTlsConfigurationResp().GetClientTlsConfiguration()
+
+	var cert tls.Certificate
+	for _, v := range tlsConfig.CertificateChain {
+		// Populate Certificates field
+		block, _ := pem.Decode([]byte(v))
+		if block == nil {
+			return nil, errors.New("certificate in CertificateChain obtained from S2Av2 is empty.")
+		}
+		x509Cert, err := x509.ParseCertificate(block.Bytes)
+		if err != nil {
+			return nil, err
+		}
+		cert.Certificate = append(cert.Certificate, x509Cert.Raw)
 	}
 
+	// Until TODO resolved, populate PrivateKey field using leaf cert and
+	// corresponding key.
+	// TODO(rmehta19): Call remote signer library for private key.
+	tlsCert, _ := tls.X509KeyPair([]byte(tlsConfig.CertificateChain[0]), clientKey)
+	cert.PrivateKey = tlsCert.PrivateKey
+
+	// TODO(rmehta19): Remove certPool when RPC containing
+	// ValidatePeerCertificateChainReq implemented.
 	rootCertPool := x509.NewCertPool()
 	rootCertPool.AppendCertsFromPEM(serverCert)
 
-	// TODO(rmehta19): Call S2Av2 for config values.
+	minVersion, maxVersion := getTLSMinMaxVersionsClient(tlsConfig)
+	if minVersion > maxVersion {
+		return nil, errors.New("S2Av2 provided minVersion > maxVersion.")
+	}
+
 	// Create mTLS credentials for client.
 	return &tls.Config {
+		// TODO(rmehta19): Make use of tlsConfig.HandshakeCiphersuites /
+		// RecordCiphersuites.
 		Certificates: []tls.Certificate{cert},
 		VerifyPeerCertificate: certverifier.VerifyServerCertificateChain("s2a_test_cert", serverHostname, rootCertPool),
 		RootCAs: rootCertPool,
 		ServerName: serverHostname,
 		InsecureSkipVerify: true,
 		ClientSessionCache: nil,
-		MinVersion: uint16(tls.VersionTLS13),
-		MaxVersion: uint16(tls.VersionTLS13),
-	}
+		MinVersion: minVersion,
+		MaxVersion: maxVersion,
+	}, nil
 }
 
 // GetTlsConfigurationForServer returns a tls.Config instance for use by a server application.
-func GetTlsConfigurationForServer() *tls.Config {
-	// TODO(rmehta19): Call S2Av2 for certificate.
-	// TODO(rmehta19): Call remote signer library for private key.
-	cert, err := tls.X509KeyPair(serverCert, serverKey)
+func GetTlsConfigurationForServer(cstream s2av2pb.S2AService_SetUpSessionClient) (*tls.Config, error) {
+	// TODO(rmehta19): move call to S2Av2 to a helper function.
+	// Send request to S2Av2 for config.
+	err := cstream.Send(&s2av2pb.SessionReq {
```

Comment:
micro-nit: here as well, please use the `if err := ...; err != nil { ... }` style